src/Stack/Upload.hs

```diff
@@ -67,14 +84,26 @@ instance FromJSON (FilePath -> HackageCreds) where
 withEnvVariable :: Text -> IO Text -> IO Text
 withEnvVariable varName fromPrompt = lookupEnv (T.unpack varName) >>= maybe fromPrompt (pure . T.pack)
 
+maybeGetHackageKey :: RIO m (Maybe HackageKey)
+maybeGetHackageKey = fmap (HackageKey . T.pack) <$> (liftIO $ lookupEnv (T.unpack "HACKAGE_KEY"))
```

Comment:
You could do the whole thing in `liftIO` and so move the `<$>` and `$` to drop one parens.

`OverloadedStrings` means the term `"HACKAGE_KEY"` can be either a `String` or `Text`, so treating it as `Text` and `unpack`ing it is functional but unnecessary. You can just treat it like a `String`.

```suggestion
maybeGetHackageKey = liftIO $ fmap (HackageKey . T.pack) <$> lookupEnv "HACKAGE_KEY"
```

I don't think `Stack.Prelude` has `(<$$>)`, but if it did you could do:

```hs
maybeGetHackageKey = liftIO $ HackageKey . T.pack <$$> lookupEnv "HACKAGE_KEY"
```